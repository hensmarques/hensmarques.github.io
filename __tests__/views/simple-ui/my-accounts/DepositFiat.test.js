import React from 'react';
import { DepositFiat } from '../../../../src/views/simple-ui/my-accounts/DepositFiat';

describe('<DepositFiat/>', () => {
  const mockProps = {
    deposit: { request: 'PENDING' }
  };

  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<DepositFiat {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('dispatches makeDeposit when form is submitted', () => {
    const makeDeposit = jest.fn();
    const mockCurrency = { id: 'BTC' };
    const mockFormData = { fullName: 'foobar', amount: '5', memo: 'biz' };
    const wrapper = shallow(
      <DepositFiat
        {...mockProps}
        currency={mockCurrency}
        makeDeposit={makeDeposit}
      />
    );
    const onSubmitSpy = jest.spyOn(wrapper.instance(), 'onSubmit');
    wrapper.instance().forceUpdate();
    wrapper.find('Connect(FormContainer)').simulate('submit', mockFormData);
    expect(onSubmitSpy).toHaveBeenCalledWith(mockFormData);
    expect(makeDeposit).toHaveBeenCalledWith(
      mockCurrency.id,
      mockFormData.amount
    );
  });

  it("dispatches showModal('none') when modal closes", () => {
    const showModal = jest.fn();
    const wrapper = shallow(
      <DepositFiat {...mockProps} showModal={showModal} />
    );
    wrapper
      .find('Connect(FormContainer)')
      .props()
      .buttons[0]['onClick']();
    expect(showModal).toHaveBeenCalledWith('none');
  });
});
