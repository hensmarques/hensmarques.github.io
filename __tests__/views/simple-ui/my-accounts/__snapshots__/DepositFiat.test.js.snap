// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DepositFiat/> renders and matches the snapshot 1`] = `
<Fragment>
  <p>
    DEPOSIT.BANK_USD
  </p>
  <p>
    DEPOSIT.BANK_USD
  </p>
  <p>
    DEPOSIT.BANK_USD
  </p>
  <p>
    DEPOSIT.BANK_USD
  </p>
  <Connect(FormContainer)
    buttons={
      Array [
        Object {
          "name": "Close",
          "onClick": [Function],
        },
      ]
    }
    className="deposit-fiat-form"
    fields={
      Array [
        Object {
          "name": "Amount",
          "required": true,
          "translation": "AMOUNT",
          "type": "text",
          "validations": Object {
            "onBlur": Array [
              Object {
                "name": "amount",
              },
            ],
          },
        },
      ]
    }
    key="deposit-fiat-form"
    onSubmit={[Function]}
    submitText="Submit"
    translation="DEPOSIT_FIAT_MODAL"
  />
</Fragment>
`;
